docs/nesting.rst

```diff
@@ -3,18 +3,49 @@ Nesting
 
     Come back to this document when you get confused about how data is stored in sockets.
 
-Sverchok sockets can contain the elements of any number of objects. The number of objects is shown by the number beside the socket name. If you look at all the previous images, you'll notice most sockets have been outputting ``socketname. 1`` (a collection, containing one thing)
+Sverchok sockets can contain the elements of any number of objects. The number of objects is shown by the number beside the socket name. If you look at all the previous images, you'll notice most sockets have been outputting ``socketname. 1`` (a collection, containing one thing).
 
-Let's look at some examples, (not necessarily outputting just one thing):
+Each collection is a list of objects and these objects are themselves lists down to the most basic level which is the list of coordinates of each vertex.
+
+So a vertex is a list of coordinates (X,Y,Z): 
+
+``v0 = (1.0, 3.8, 2.5) v1 = (0.0, 0.0,0.0) v2 = (2.1, 3.1, 4.1)`` 
+
+A list of vertexes is made of several coordinates: 
+
+``vertices = [(1.0, 3.8, 2.5), (0.0,0.0,0.0), (2.1,3.1,4.1)]``
+
+And each list element has an index starting by "0" and ending in "n", from left to right. So, in the above vertices list the indices for each vertex are:
+
+``v0 = 0, v1 = 1, v2 = 2``
+
+If, for example, you want to use the vertices to generate edges, you'll need to call upon their indexes to generate each edge. If we want to generate "edge 0" based on vertices v1 and v2 we will call them using index 1 and index 2 from that list, we will be creating an edge list like this: 
+
+``edge = (1 ,2)``
+
+If we want to generate an object composed by 3 edges from the above vertices indices we will need to figure out to retrieve the indices of the vertices we need and to create a list that might look like this:
+
+``edges_list = [(0, 1), (1, 2), (2, 3)]``
+
+So basically, sverchok works by using nodes to creat lists of values, then using other nodes to evaluate those lists and generate new lists. Eventually the results of the final list will be used to generate the final output that will be passed to the viewport with a viewer node.
+
+This is why the nesting concept is fundamental in order to use Sverchok.
+
+Let's look at some examples, (not necessarily outputting just one thing): 
+
+NOTE to zeffii, what do you mean with this: (not necessarily outputting just one thing)?
```

Comment:
num things | descriptor
----|----
"one thing" | would be a socket that outputs a single collection, so `[[item1, item2, item3]]`
"two things" | would be a socket that outputs two collections `[[item1, item2, item3], [item1, item2, item3]]`

Reply:
fair comment btw, if that line didn't make sense on the first reading. The comment in parens could be dropped, i don't think it was helpful the way i phrased it originally.